### PR TITLE
solve the task

### PR DESCRIPTION
**Pull request by esidarapp**

### CODE REVIEW

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.example.spring.service;
+
+import com.example.spring.model.Book;
+import com.example.spring.repository.BookRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookServiceImpl.java:13]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import com.example.spring.model.Book;
+import com.example.spring.repository.BookRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired
+    private BookRepository bookRepository;
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookServiceImpl.java]:
```suggestion
    private final BookRepository bookRepository;
```

File: src/main/java/com/example/spring/service/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.example.spring.model.Book;
+import com.example.spring.repository.BookRepository;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BookServiceImpl implements BookService {
+    @Autowired
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/service/BookServiceImpl.java]:
```suggestion
```

File: src/main/java/com/example/spring/repository/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+            return book;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't insert book into DB: " + book);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert book into DB: " + book);

```
It's better to create custom exception

File: src/main/java/com/example/spring/repository/BookRepositoryImpl.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        }
+    }
+
+    @Override
+    public List<Book> findAll() {
+        try (Session session = sessionFactory.openSession()) {
+            Query<Book> query = session.createQuery("FROM Book",Book.class);
+            return query.getResultList();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't get all books", e);
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/BookRepositoryImpl.java]:
```suggestion
            throw new DataProcessingException("Can't get all books", e);

```

File: src/test/resources/application.properties [hunk 6]
```diff
@@ -0,0 +1,5 @@
+spring.datasource.url=jdbc:h2:mem:testdb
+spring.datasource.driverClassName=org.h2.Driver
+spring.datasource.username=sa
+spring.datasource.password=password
+spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
```
Comment (Elena-Bruyako) [inline src/test/resources/application.properties:5]:
```suggestion
spring.jpa.database-platform=org.hibernate.dialect.H2Dialect

```

File: src/main/java/com/example/spring/repository/BookRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+import com.example.spring.model.Book;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/BookRepositoryImpl.java:13]:
```suggestion
@Repository
@RequiredArgsConstructor
```

File: src/main/java/com/example/spring/repository/BookRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+public class BookRepositoryImpl implements BookRepository {
+    private final SessionFactory sessionFactory;
+
+    @Autowired
+    public BookRepositoryImpl(SessionFactory sessionFactory) {
```
Comment (Elena-Bruyako) [inline src/main/java/com/example/spring/repository/BookRepositoryImpl.java]: remove constructor

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

File: src/main/resources/application.properties [hunk 9]
```diff
@@ -1 +1,11 @@
...
+
+spring.datasource.url=jdbc:mysql://localhost:3306/library
+spring.datasource.username=root
+spring.datasource.password=timofeev
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQLDialect
+spring.jpa.show-sql=true
+spring.jpa.hibernate.ddl-auto=create-drop
+
```
Comment (okuzan) [inline src/main/resources/application.properties]:
```suggestion
```

**Review by okuzan — APPROVED** [review]

**Comment by Elena-Bruyako** [comment]
Screenshot 2024-06-11 at 08 18 54
Your changes were approved, create new PR with HW implementation and send it, don't mix up your HWs in one PR

**Pull request merged**